---
output:
  pdf_document: default
  html_document: default
---
# Appendices

```{r, include=FALSE}
knitr::opts_chunk$set(cache = TRUE)
```

**Word To The Wise** 

- LEARN GIT, get a Github account! Put Every computer program you write and Eevery stick of knowledge related to your work in a clean format on Git. You will benefit from it in the long run...

- To Biologists and Biochemists (which I also consider myself), it is your task to become more familiar with computer languages, miscellaneous computer concepts and statistics. 

As of the writing of this booklet, Data Science and Bioinformatics are now centered around the computer languages [R](https://cran.r-project.org/) and [Python](https://www.anaconda.com/).  

To many researchers in data science and bioinformatics the field now includes such languages as,  
*in no particular order*,

- [R](https://cran.r-project.org/)
- [Python](https://www.anaconda.com/)  
- [Bash shell scripting](https://www.datascienceatthecommandline.com/)  
- [SQL](https://en.wikipedia.org/wiki/SQL)  
- [Julia](https://julialang.org/)  
- [Perl](https://www.perl.org/) 
- [C](https://en.wikipedia.org/wiki/C_%28programming_language%29)  
- [Javascript](https://www.javascript.com/) 
- [RMarkdown](https://rmarkdown.rstudio.com/), (this one is fun and easy)

#### Just start with one language {-} 

- Stick with it for a time and learn it. Learn the ins-and-outs of one language first.

#### Is R or Python better? {-}

1. R & Python are both FREE,
1. Both have great integrated development environments (IDEs),
   - [RStudio](https://www.rstudio.com/) is great & FREE,
   - [Spyder](https://www.spyder-ide.org/) is also great & FREE,
   - [PyCharm](https://www.jetbrains.com/pycharm/)
   - [Sublime](https://www.sublimetext.com/)
   - [Visual Studio Code](https://code.visualstudio.com/)
   - [Vim](https://www.vim.org/)
   - [Jupyter Notebook](https://jupyter.org/)
1. Both languages have been around for > 20 years, therefore both have tons of FREE information & tutorials on YouTube,

## Install R & RStudio

**NOTE**: I use [Ubuntu](https://www.ubuntu.com/).

1. Go to: https://cran.r-project.org/

1. Choose R for your operating system.

1. If you are using Linux, I recommend that you download/install 4 R files.
   1. r-base-core_#.#.#.*.deb  (approx. 30 MB)
   1. r-base-dev_#.#.#.*.deb   (approx. 45 KB)
   1. r-base_#.#.#*.deb        (approx. 90 KB)
   1. r-base-html-#.#.#*.deb   (approx. 90 KB)

1. If you do have Linux you may try this video: [How to install R.](https://www.youtube.com/watch?v=LeB15G14N7g)
   
1. Go to: https://www.rstudio.com/

1. From RStudio's homepage click, *Products* then click *RStudio* from the dropdown menu.

1. Click the Download of the FREE version of RStudio Desktop

1. Click *RStudio #.#* to download a version for your machine

1. Have Linux? Try this - [How to install RStudio.](https://www.youtube.com/watch?v=LeB15G14N7g)

1. If you are looking for instructions for Mac & Windows machines try: 
   - [FreeCodeCamp](https://guide.freecodecamp.org/r/)
   
## Load Libraries Used In This Project
 
If you are using Ubuntu/Linux you may need these Linux libraries first.

```
sudo apt-get install libcurl4-openssl-dev
sudo apt-get install libssl-dev
sudo apt-get install libxml2-dev
```

To install `car` & `rgl`

```
sudo apt-get install xorg
sudo apt-get install libx11-dev
sudo apt-get install libglu1-mesa-dev
sudo apt-get install libfreetype6-dev
```


```
install.packages("rlang")
library(rlang)

load_or_install <- function(package_names) {
    for(package_name in package_names) {
        if(!is_installed(package_name)) {
            install.packages(package_name,
            repos = "http://lib.stat.cmu.edu/R/CRAN",
            dependencies = TRUE)
        }
        library(package_name,character.only=TRUE,quietly=TRUE,verbose=FALSE) 
        print("OK")
        }
}
```

```
load_or_install(c("readr", "doMC", "corrplot", "knitr", "caret", "tidyverse"))

load_or_install(c("ggplot2", "rmarkdown", "bookdown", "blogdown", "kernlab"))

load_or_install(c("e1071", "plyr", "RColorBrewer", "neuralnet"))

load_or_install(c("rpart", "MASS", "tidyr", "ggplot2", "seqinr"))

load_or_install(c("LogicReg", "randomForest", "foreach"))

load_or_install(c("import", "dplyr", "stringr", "stringi"))
```

## Calculate the amino acid compositions (AAC) and Di-peptide compositions (DPC)

 from .fasta formats, {Myoglobin, Non-Myoglobin}

Calculating the Amino Acid and Di-peptide composition of a protein string is a 
simple calculation requiring the total amino acid length of the peptide or 
poly-peptide of interest and a count of substrings. Initially, the command 
`seqinr::read.fasta` reads .fasta file formats and returns a list of proteins 
stripping away all other information. Secondly, the command 
`stringr::str_count()` produces an integer value of the number of substrings 
in a larger string, i.e. `peptide`. 

For example, `aa_nums[j] = str_count(peptide, col_titles[j]) / total_aa`, 

Where; `aa_nums[j]` is an array to saving values for later writing to file, 
`peptide` is the string to check, i.e. protein of interest, `col_titles[j]` 
is the substring which is either a single amino acid or di-peptide.

Input: .fasta  
Output: .csv

Libraries 

```
Libraries = c("stringr", "knitr", "seqinr")

for (p in Libraries) {  # Install Libraries
    library(p, character.only = TRUE)
}

opts_chunk$set(cache = TRUE, 
               warning = FALSE, 
               message = FALSE,
               align = "center")
```

Import uniprot-myoglobin.fasta - Read peptide lines

```
read_fasta <- function(file) {
    listo_proteins <- read.fasta(file = file,
                                 seqtype = "AA",
                                 as.string = TRUE,
                                 seqonly = FALSE,
                                 strip.desc = TRUE)
    return(listo_proteins)
}

file = "./00-data/ORIGINAL_DATA/uniprot-myoglobin.fasta"
myoglobins <- read_fasta(file)
```

Column_titles
```
column_titles = function() {
    peptides = c("A", "C", "D", "E", "F",
                 "G", "H", "I", "K", "L",
                 "M", "N", "P", "Q", "R",
                 "S", "T", "V", "W", "Y")
                 
    # Add DIPEPTIDES column titles
    di_titles = vector(mode = "character", length = 400)
    k = 1
    for (i in 1:20) {
        for (j in 1:20) {
            di_titles[k] <- paste(peptides[i], peptides[j], sep = "")
            k = k + 1
        }
    }
    aa_di_titles <- c("Class","TotalAA","PID", peptides, di_titles)
    return(aa_di_titles)
}

col_titles <- column_titles()
col_titles
```

Write empty .csv

```
write_empty_csv <- function(protein_class = "C") {
    col_titles <- column_titles()
    file_name <- paste(protein_class, "_aac_dpc.csv", sep = "")
    write.table(t(col_titles), 
                file_name,  
                sep = ",", 
                col.names = FALSE, 
                row.names = FALSE, 
                eol = "\n")
    return(file_name)
}

file_name <- write_empty_csv()
```

## Calculate AAC and DPC values function

```
calc_aac_dpc <- function(peptide, protein_class = "C", i, file_name) {
    aa_nums = matrix(0, ncol = 423)
    ###############################
    # First column is class
    aa_nums[1] = ifelse(protein_class == "C", 0, 1)
    # Second column is total number of amino acids
    total_aa = nchar(peptide)
    aa_nums[2] = total_aa
    # Third line is 'Protein ID', PID
    aa_nums[3] = paste(protein_class, i, sep = "")
    # Column 4:423 - Calculate AAC/DPC
    for (j in 4:423) {
        aa_nums[j] = str_count(peptide, col_titles[j]) / total_aa
        }
    write(t(aa_nums), file = file_name, append = TRUE, ncolumns = 423, sep = ",")
}
```

## Run Myoglobin

```
# RUN Myoglobin
for (i in 1:1124) {
    peptide <- myoglobins[[i]][1]
    calc_aac_dpc(peptide, protein_class = "M", i, file_name)
}
```

## Run Control / Human-NOT-myoglobin 

- Import data - Read peptide lines

```
read_fasta <- function(file) {
    listo_proteins <- read.fasta(file = file,
                            seqtype = "AA",
                            as.string = TRUE,
                            seqonly = FALSE,
                            strip.desc = TRUE)
    return(listo_proteins)
}

file = "./00-data/ORIGINAL_DATA/uniprot-human+NOT+hemoglobin+NOT+myoglobin+random.fasta"
controls <- read_fasta(file)
```

## Run Controls

```
for (i in 1:1216) {
    peptide <- controls[[i]][1]
    calc_aac_dpc(peptide, protein_class = "C", i, file_name)
}
```

## KEEP AAC ONLY FOR RAW DATA

```
file = "./00-data/aac_dpc_values/C+M_aac_dpc.csv"
C+M_aac_dpc <- read.csv(file,
                        stringsAsFactors=FALSE)
# View(`C+M_aac_dpc`)

# Select 1st thru 23rd variables
c_m_RAW_AAC <- C+M_aac_dpc[c(1:23)]
```

- To A Comma Delimited Text File

```
setwd("../00-data/02-aac_dpc_values/")

write.table(c_m_RAW_AAC, 
            file = "./00-data/02-aac_dpc_values/c_m_RAW_AAC.csv", 
            sep = ",", 
            row.names = F)
```

## Transform {C, F, I} from c_m_RAW_AAC

```
library(readr)

file = "../00-data/02-aac_dpc_values/c_m_RAW_AAC.csv"
c_m_RAW_AAC <- read_csv(file, 
                        col_types = cols(Class = col_factor(levels = c("0","1"))))
c_m_TRANSFORMED_AAC <- c_m_RAW_AAC
```

1. Transfrom C,F,I using sqrt(x)  
2. Columns: C=5, F=8, I=11  

```
c_m_TRANSFORMED_AAC[, 5] <- sqrt(c_m_TRANSFORMED_AAC[, 5]) # C
c_m_TRANSFORMED_AAC[, 8] <- sqrt(c_m_TRANSFORMED_AAC[, 8]) # F
c_m_TRANSFORMED_AAC[,11] <- sqrt(c_m_TRANSFORMED_AAC[,11]) # I

file = "./00-data/02-aac_dpc_values/c_m_TRANSFORMED.csv"
write_csv(c_m_TRANSFORMED_AAC, 
          file = file, 
          col_names = T)
```

## Where To Find Help

1. [Cheat Sheets](https://resources.rstudio.com/rstudio-cheatsheets)
1. https://community.rstudio.com
1. https://www.reddit.com/r/RStudio/
1. https://R-bloggers.com/
1. https://resources.rstudio.com/
1. [Rpubs.com](https://rpubs.com/)

- **Rpubs.com** contains R/RStudio notebooks and Markdown pages, VERY HELPFUL work from other peoples online R documents. It is a way to learn from others and share your work. - Sign up, it is [FREE](https://rpubs.com/)! then press: *Get Started*

**NOTE**: If you are interested in seeing what others have published search Google, Rpubs.com does not have its own search function. In Google, Search: `site:rpubs.com eda`

Other sites:  

1. [Coursera](https://www.coursera.org/)
1. [Stack Overflow](https://stackoverflow.com/)
1. [Quora](https://www.quora.com/What-is-data-science)
1. [Roger Peng's EDA](https://youtu.be/ZXrHkIz-krE?t=15)  
1. [Bookdown](https://bookdown.org/) - **terrible** not necessary resource  

The Lean Publishing (https://leanpub.com) company contains a library in the form of FREE down-loadable books/pdfs. I recommend;

1. How to be a modern scientist[^91]  by Jeffrey Leek[^92] 
2. R Programming for Data Science[^93]  by Roger Peng[^94] 
3. Exploratory Data Analysis with R[^95]  by Roger Peng
4. Data Analysis for the Life Sciences[^96]  by Rafael Irizarry & Michael Love

[^91]:https://leanpub.com/modernscientist
[^92]:http://jtleek.com
[^93]:https://leanpub.com/rprogramming
[^94]:https://simplystatistics.org
[^95]:https://leanpub.com/exdata
[^96]:https://leanpub.com/dataanalysisforthelifesciences

## Machine Setting & Session Info

```{r, include=FALSE}
library(knitr)
opts_chunk$set(cache = TRUE, warning = FALSE, message = FALSE)
```

```{r}
Sys.info()[c(1:3,5)]
```

```{r}
sessionInfo()
```

